### PR TITLE
Feature/ops cg fortran

### PR DESCRIPTION
**Pull request by Ashutosh-Londhe**

### CODE REVIEW

**Review by benielT — COMMENTED** [review]
I have added some comments. Generally, changes are good.

File: ops/c/include/ops_internal2.h [hunk 1]
```diff
@@ -139,19 +139,20 @@ struct ops_kernel {
 
 /** Storage for OPS parallel handles */
 struct ops_kernel_descriptor {
-  const char *name;           /**< name of kernel */
+  char *name;           /**< name of kernel */
+  int name_len;         /**< kernel name length */
```
[the remark this answers is not in the document]
Comment (Ashutosh-Londhe) [reply ops/c/include/ops_internal2.h:143]: this is required in fortran side when converting pointer from C to F

File: ops/c/src/core/ops_lazy.cpp [hunk 2]
```diff
@@ -937,12 +947,56 @@ void ops_execute(OPS_instance *instance) {
   ops_kernel_list.clear();
 }
 
+void create_kerneldesc_and_enque(char const *name, char const* kernel_name, ops_arg *args, int nargs, int index, int dim, int isdevice, int *range, ops_block block, void (*func)(struct ops_kernel_descriptor *desc))
```
[the remark this answers is not in the document]
Comment (Ashutosh-Londhe) [reply ops/c/src/core/ops_lazy.cpp:950]: yes, this was added as part of fortran-c interface so using it directly in C code gen also

**Review by benielT — APPROVED** [review]
Seems good.

**Pull request merged**